### ISSUE TITLE
Affine Noise (as well as non parameterized noise)

### ISSUE THREAD

**Issue by Gaussia** [JuliaDiffEq/DiffEqBiological.jl#19]
By default the DSL generates noise term likes:
```julia
g[n,m] = stoch_nm * sqrt(rate_m)
```
It is possible the declare a special noise scaling parameter eta so that the term rather becomes
```julia
g[n,m] = eta * stoch_nm * sqrt(rate_m)
```
An improvement as mentioned in the previous PR would be to have affine noise a la
```julia
g[n,m] = eta1 * stoch_nm * sqrt(rate_m) + eta2
```
(I hope that I understood this right)

the new call would be
```julia
rn = @reaction_network rType (eta1,eta2) begin
    ...
end
```
alternatively
```julia
rn = @reaction_network rType begin
    ...
end
```
if no noise customization is desired.

On the same topic. Now noise is automatically stored as a parameter (default the last parameter in the parameter array). This is advantageous since you can the change the noise rate without redoing the DSL call. However in some cases you want a fixed value. I think it would be advantageous to allow for number as well, like:
```julia
rn = @reaction_network rType (2.0,eta2) begin
    ...
end
```
now the multiplicative scaling would be 2.0, with no changing that. This can be especially usefull if you are only interested in using parts of the affine noise, like:
```julia
rn = @reaction_network rType (0.0,eta2) begin
    ...
end
```

**Comment by Gaussia** [comment]
Maybe that would be an alternative solution and we could drop this.

One slight comment however, if one solely relied on such an AffineNoiseProblem then one would lose the correlation in noise. One could make the overload such that one will get the correlated affine noise, however then one would not be able to e.g. use you improved method?

**Comment by ChrisRackauckas** [comment]
>One slight comment however, if one solely relied on such an AffineNoiseProblem then one would lose the correlation in noise. One could make the overload such that one will get the correlated affine noise, however then one would not be able to e.g. use you improved method?

Yes, it loses the correlations in the noise. That's the tradeoff that has to be made there. In many cases though, noise is phenomenological, so it's "the ODE + some randomness". Of course, it's missing some of the reaction structure, but the speed can be much better than the full non-diagonal SDE.

**Comment by Gaussia** [comment]
Basically, it is no pressing need, but it doesn't really hurt to have the option either.

**Comment by Gaussia** [comment]
So I have been thinking some more.

The `AffineNoiseProblem` have not been added yet? Still, assuming that it does get added then I am unsure whenever there is a reason for adding affine noise here as well.

Multiplicative noise should still be there, it can be useful if one have some model with concentrations around 1 and still want to use proper coupled noise, but need to scale it down a little. 

However I am unsure in what situation one would be interested in having proper coupled noise, scale it a little and then add som additive noise as well. If what one wants is only "the ODE + some randomness"  then it would be better to just make an  `AffineNoiseProblem` and then get access to much faster solvers (when/if they are available). If one is using non-diagonal noise anyway just adding some extra additive randomness seems odd, although someone at some point might want to do it.

I might add this option anyway, however I start to think that making something like
```julia
rn = @reaction_network rType (eta1,eta2) begin
    ...
end
```
the standard notation is unnecessary. I might add it as an option anyway, but I think it is better if one can simply add a multiplicative factor to the non-diagonal noise like
```julia
rn = @reaction_network rType eta begin
    ...
end
```

If I do add multiple dispatch so that it is possible to declare noise multiplication without declaring a special type I will probably also force the noise term to be declared among the parameters like:
```julia
rn = @reaction_network eta begin
    ...
end eta
```

**Comment by ChrisRackauckas** [comment]
Not added yet. I'm wondering if it's better to handle the transform in ModelingToolkit.jl.

>However I am unsure in what situation one would be interested in having proper coupled noise, scale it a little and then add som additive noise as well. If what one wants is only "the ODE + some randomness" then it would be better to just make an AffineNoiseProblem and then get access to much faster solvers (when/if they are available). If one is using non-diagonal noise anyway just adding some extra additive randomness seems odd, although someone at some point might want to do it.

No correlations or coupling, just diagonal multiplicative.

**Comment by Gaussia** [comment]
> No correlations or coupling, just diagonal multiplicative.
Yes. Since there is no real reason to do the thing I outlined there is no need to add support to it (and we only keep the part were one can scale the correlated/coupled noise multiplicative). Diagonal multiplicative with a possible additive term will be fixed by the AffineNoiseProblem, whenever wherever it is added.

**Comment by ChrisRackauckas** [comment]
The paper describing the affine noise stuff is on Arxiv:

https://arxiv.org/abs/1804.04344

Basically, if you don't need to model the noise correlations you can get a lot faster than the current non-diagonal tools. But we can hold off on this for now since we don't have a good way to transform affine noise into additive noise. ModelingToolkit.jl will help here, but we're not there yet.

**Comment by Gaussia** [comment]
Looks good, looking forward to be able to test it.
